### PR TITLE
KAFKA-16231: Update consumer_test.py to support KIP-848’s group protocol config

### PR DESCRIPTION
**Pull request by kirktrue**
Added a new optional `group_protocol` parameter to the test methods, then passed that down to the `setup_consumer` method.

Unfortunately, because the new consumer can only be used with the new coordinator, this required a new `@matrix` block instead of adding the `group_protocol=["classic", "consumer"]` to the existing blocks 😢

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by lucasbru** [comment]
looks good. As you said, we shouldn't merge this yet as long as some are not passing. Could we follow a similar approach as with the integration tests, enable the ones that are passing, and leave the failing ones disabled, ideally linking a ticket?

**Comment by kirktrue** [comment]
@lucasbru can you take another look at this PR? Thanks!

**Comment by kirktrue** [comment]
> Could we follow a similar approach as with the integration tests, enable the ones that are passing, and leave the failing ones disabled, ideally linking a ticket?

Yes, this is the approach I took. Right now, the `all_group_protocols` list (defined in `consumer_group.py`) is—for the time being—just a list of one: the "classic" consumer.

File: tests/kafkatest/tests/client/consumer_test.py
Comment (lucasbru) [inline tests/kafkatest/tests/client/consumer_test.py]: Why no group protocol parameter here?

File: tests/kafkatest/tests/verifiable_consumer_test.py
Comment (lucasbru) [inline tests/kafkatest/tests/verifiable_consumer_test.py]: Why do we have to do this? Are we still checking the same thing if we consider "started" as "joined"?

**Review by lucasbru — COMMENTED** [review]
Hey @kirktrue, thanks for PR!
 - You changed some of the testing code. Are these just refactorings? Please leave some comments why we are changing the code around waiting for states and assignments. I don't want to stop you from cleaning up things, I just have a hard time following the changes.
 - Why do we only add the group protocol parameter in some of the tests?
 - Why do you want to commit this without adding the `consumer` group in the set of valid group protocols. Are none of the tests passing right now? Feels like we are adding dead code.

**Review by lucasbru — APPROVED** [review]

**Comment by lucasbru** [comment]
LGTM, thanks!

**Pull request merged**
